R/parallel-simulation-task.R

```diff
@@ -0,0 +1,53 @@
+#' @title ParallelSimulationTask
+#' @description  R6 class for ParallelSimulationTask settings
+ParallelSimulationTask <- R6::R6Class(
+  "ParallelSimulationTask",
+  inherit = SimulationTask,
+  public = list(
+
+    #' @description
+    #' Create a `ParallelSimulationTask` object
+    #' @param ... parameters inherited from R6 class `SimulationTask` object
+    #' @return A new `SimulationTask` object
+    initialize = function(...) {
+      super$initialize(...)
+    },
+
+    #' @description
+    #' Run task and save its output
+    #' @param structureSets list of `SimulationStructure` R6 class
+    runTask = function(structureSets) {
```

Comment:
I don't see where stuff is actually being run in //. Where is runSimulations called?

Reply:
The parallel simulation takes place in the `taskFunction` for this task, which is assigned in the function `loadSimulateTask`.  If a mean model workflow is to be simulated (in parallel) then the task function is `simulateModelParallel` (called from the `runTask` function of a `ParallelSimulationTask` object.  Otherwise the population simulation sets are run sequentially thru the `simulateModelForPopulation` function, which is called in a for loop (over all simulation sets) in the `runTask` function of a `SimulationTask` object.

Reply:
ok thanks